lang/src/org/partiql/lang/eval/ExprValueExtensions.kt

```diff
@@ -77,8 +77,11 @@ fun ExprValue.numberValue(): Number =
 fun ExprValue.timestampValue(): Timestamp =
     scalar.timestampValue() ?: errNoContext("Expected timestamp: $ionValue", internal = false)
 
-fun ExprValue.stringValue(): String =
-    scalar.stringValue() ?: errNoContext("Expected text: $ionValue", internal = false)
+fun ExprValue.stringValue(): String = scalar.stringValue() ?:
```

Comment:
This is a general extension function that is called in multiple use cases that require to obtain a `String` from an `ExprValue`, one such case is when we need to evaluate a string append `||` operator. 

Making this extension function report an error that *always* states it is due to a string append operation might lead to miscommunication. 

I think the logic needs to go [here](https://github.com/partiql/partiql-lang-kotlin/blob/1ad2e93c1192d79029f22f687587d362eb328454/lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt#L516) where the code deals with string append. Each argument to `||` needs to be checked and if it is not a `String` value then report the more specific error message that `||` did not receive a value of the appropriate type (`String`)

Reply:
Makes sense - I'll attempt to fix it before next week.

Reply:
@therapon Can anyone take this over? A lot going on in my life, I would love to help you out but I cannot do that. Should I just trash this PR?

Reply:
No problem, I'm glad I could contribute something meaningful.